woocommerce/woocommerce
# Change the custom field name component to be an autocomplete

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #46919



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure `Try the new product editor (Beta)` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure `product-custom-fields` feature toggle is on under the under the `WooCommerce Admin Test Helper` plugin features tab.
3. Go to `Products > Add new` and click the `Organization` tab
4. Switch on the `Show custom fields` toggle.
5. The `Custom fields` section should be shown.
6. Click `Add new` button.
7. From the `Add custom fields` modal, the custom field name should now be an autocomplete, from there should be possible to create a new name or find and existing one.
8. The same as 7 should be possible when editing the custom field in its own edition modal.
9. Validations should remains as before this PR.
10. Note that the behavior of the autocomplete is based on the [ComboboxControl](https://wordpress.github.io/gutenberg/?path=/docs/components-comboboxcontrol--docs) from core.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
